### ISSUE TITLE
Error building ConcaveMesh with TriangleVertexArray

### ISSUE THREAD

**Issue by Josh-Gibson** [DanielChappuis/reactphysics3d#285]
Hello! This problem has stumped me, and I am hoping to get some guidance.
So I don't believe this is a bug, but I can't seem to get this set up right.

I wish to create a ConcaveMeshShape with floats and vertices I generate. I am generating this from learnopengl's heightmap from an image:
https://learnopengl.com/Guest-Articles/2021/Tessellation/Height-map

I managed to get it so that I have the concave mesh generated, yet it has some extra triangles that not only aren't part of the shape, but seem to make the program crash more often.

It looks similar to this issue: #262  

This is the result from rendering the collisionshape:

![error](https://user-images.githubusercontent.com/22622013/199386876-6252ae7b-dfde-4407-bf27-87e1775d94ae.png)

The heightmap above is correct. I have loaded different images to create different shapes and they all work. The issue is those flat lines at the bottom. I don't know how to remove those. If I load less indices or vertices, the shape is cut but the lines are still there.

I have also looked in ConcaveMesh.cpp, but it seems right so I'm not sure what I'm doing wrong

There are only two areas where I think I am messing up: creating the vertices and indices, or creating the triangle vertex array. Everything else works so far (collisions, rendering colliders)

My code for making the heightmap is the same as the one from learnopengl. Here's how I generate the vertices and indicies.:

```
for(unsigned int i = 0; i < hheight; i++)
	{
		for(unsigned int j = 0; j < wwidth; j++)
		{
			// retrieve texel for (i,j) tex coord
			unsigned char* texel = data + (j + wwidth * i) * nChannels;
			// raw height at coordinate
			unsigned char y = texel[0];
			// vertex
			vertices.push_back( -hheight/2.0f +i);        // v.x
			vertices.push_back( (int)y * yScale - yShift); // v.y
			vertices.push_back( -wwidth/2.0f + j);        // v.z
		}
	}
	
	std::vector<int> indices;
	for(unsigned int i = 0; i < hheight-1; i++)       // for each row a.k.a. each strip
	{
		for(unsigned int j = 0; j < wwidth; j++)      // for each column
		{
			for(unsigned int k = 0; k < 2; k++)      // for each side of the strip
			{
				indices.push_back(j + (wwidth) * (i + k));
			}
		}
	}
```

wwidth and hheight are my variables for the width and height of the image I load.

And here is how I'm setting up the trianglemesh:

```
rp3d::TriangleVertexArray* triangleArray = new rp3d::TriangleVertexArray(
		vertices.size(), 
		&vertices[0], 
		3* sizeof(float), 
		indices.size(), 
		&indices[0], 
		sizeof(int), 
		rp3d::TriangleVertexArray::VertexDataType::VERTEX_FLOAT_TYPE, 
		rp3d::TriangleVertexArray::IndexDataType::INDEX_INTEGER_TYPE);
```

&vertices[0] and &indices[0] are references to get the vector as an array.

I am unsure what the issue could be. Hopefully you can help some.

**Comment by DanielChappuis** [comment]
Hello,

When you create your TriangleVertexArray, for the 4th parameter should be the number of triangles but in your code, it seems that you use the number of indices instead (indices.size()). You probably want to use indices.size() / 3 to have the number of triangles.

I if understand correctly, you are trying to create a ConcaveMeshShape from a heighmap. You might want to to use the HeightFieldShape of the library for that. You can directly give it the height values of your heightmap. This way, you do not have to compute the mesh yourself.

**Comment by Josh-Gibson** [comment]
That doesn't work. Any sort of combination of changing the parameters either gives similar results or crashes the program. 

I believe the problem is that the way the indices are created don't line up with the triangles. I tested it by hardcoding in vertices and indices and was able to get a small result of what I wanted. I think it's because the indices keep pointing to vertices that aren't tangent to each other, and that's making the weird lines.  

The main reason why I wanted a concave shape over the heightfield shape is that I want to draw it. The heightmap I made came from learnopengl, and I was able to draw that. Does ReactPhysics have a way to get the physics object shape so that I can get things like vertices and normals to draw things like textures on it?
